main.nf

```diff
@@ -468,6 +502,64 @@ if (params.peptide_fasta){
   }
 }
 
+if (!params.skip_trimming && !params.bam){
+  process fastp {
+      label 'process_low'
+      tag "$name"
+      publishDir "${params.outdir}/fastp", mode: 'copy',
+        saveAs: {filename ->
+                    if (filename.indexOf(".fastq.gz") == -1) "logs/$filename"
+                    else if (reads[1] == null) "single_end/$filename"
+                    else if (reads[1] != null) "paired_end/$filename"
+                    else null
+                }
+
+      input:
+      set val(name), file(reads) from ch_read_files_trimming
+
+      output:
+      set val(name), file("*trimmed.fastq.gz") into ch_reads_trimmed
+      file "*fastp.json" into ch_fastp_results
+      file "*fastp.html" into ch_fastp_html
+
+      script:
+      // One set of reads --> single end
```

Comment:
Are you not using `params.single_end` for this?

We switched from this kind of implicit check to the `--single_end` flag because it’s pretty easy to mess up the glob pattern and accidentally run PE data in SE data if you’re not careful.

Reply:
ok 👌